Refactoring Advanced Features and New Sphinx Theme

Transurgeon opened the pull request:
## Description
Please include a short summary of the change.
This PR add a fresh look for the cvxpy web docs using sphinx-immaterial
Massive credit must be given to garthee who originally wrote the code [here](https://github.com/garthee/cvxpy/commit/87f47dd83dd74ab1335b7d748d9d659ec2289a15).
The website can be previewed [here](https://cvxpy.readthedocs.io/en/refactoring-advanced-features/).
I think we should merge this before making additional changes to the advanced features,
as we will be able to use sphinx-immaterial customisations.
I personally like the horizontal nav bar, but we definitely need to reduce the number of links at the top level if we decide to stick with it. 
Other changes:
- adding more dropdowns for solver options in the advanced features section.
- replacing ECOS with Clarabel on two occasions
Issue link (if applicable):
#2287 
## Type of change
- [ ] New feature (backwards compatible)
- [ ] New feature (breaking API changes)
- [ ] Bug fix
- [x] Other (Documentation, CI, ...)

## [Contribution checklist](https://www.cvxpy.org/contributing/index.html#contribution-checklist)
- [x] Add our license to new files.
- [x] Check that your code adheres to our coding style.
- [x] Write unittests.
- [x] Run the unittests and check that they’re passing.
- [x] Run the benchmarks to make sure your change doesn’t introduce a regression.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cvxpy/cvxpy?pullRequest=2289) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/cvxpy/cvxpy?pullRequest=2289) before we can accept your contribution.<br/><hr/>**Transurgeon** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cvxpy/cvxpy?pullRequest=2289) it.</sub>

Transurgeon commented:
The long awaited changes to the website are finally here. 
After some discussions with @PTNobel , I decided to reformat the initially cumbersome ``advanced features`` section in the following way. 
- moved SDP, MIP, attributes to a new section titled ``Advanced Constraints``
- moved solver related stuff to a new section titled ``Solver Features``
- moved DPP and sensitivity analysis to a new section titled ``DPP``
- combined license, related projects, citations and short course into a new top level page titled ``other``
- removed content tree in FAQ since the new theme does it automatically
- other minor changes that use the sphinx immaterial theme features (ex: using collapsible instead of dropdown)

Please let me know if the changes make sense and are according to what you envisioned.

PTNobel commented on doc/source/tutorial/advanced/index.rst:
```diff
@@ -44,6 +44,250 @@ You can use CVXPY to find the optimal dual variables for a problem. When you cal
 
 The dual variable for ``x - y >= 1`` is 2. By complementarity this implies that ``x - y`` is 1, which we can see is true. The fact that the dual variable is non-zero also tells us that if we tighten ``x - y >= 1``, (i.e., increase the right-hand side), the optimal value of the problem will increase.
 
+<<<<<<< HEAD
```
I don't think this should be here...

PTNobel commented on doc/source/tutorial/advanced/index.rst:
```diff
@@ -44,6 +44,250 @@ You can use CVXPY to find the optimal dual variables for a problem. When you cal
+* ``expr.H`` gives the Hermitian (conjugate) transpose of ``expr``.
+
+=======
+>>>>>>> c4aab172d2cb5c4a49d6a826dbd5bd50f5e02f0b
```
Similar

PTNobel reviewed (changes requested):
Couple small merge errors

Transurgeon commented:
@PTNobel I decided to create a new PR #2309 since the theme changes are still showing in this PR, and also I am not too confident in resolving the merge conflicts without making undesired changes.
